src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -921,7 +921,14 @@ const WorkflowView = _.flow(
       deleting && h(DeleteWorkflowModal, {
         workspace, methodConfig: savedConfig,
         onDismiss: () => this.setState({ deleting: false }),
-        onSuccess: () => Nav.goToPath('workspace-workflows', _.pick(['namespace', 'name'], workspace))
+        onSuccess: () => {
+          // Return to workflows list after modal is dismissed. DeleteWorkflowModal calls
+          // onDismiss after onSuccess. Thus, navigating away from this page immediately
+          // causes a warning about a state update on an unmounted component.
+          setTimeout(() => {
+            Nav.goToPath('workspace-workflows', _.pick(['namespace', 'name'], workspace))
+          }, 0)
```

Comment:
I wonder if this might be worth a utility:
```suggestion
  const onNextTick = (f, ...args) => setTimeout(() => f(...args), 0)
  // Then, in your code:
  onNextTick(Nav.goToPath, 'workspace-workflows', _.pick(['namespace', 'name'], workspace))
```

Reply:
Added in https://github.com/DataBiosphere/terra-ui/pull/2950/commits/813d77569a8277a348578f1c19fb2d4dc174646f.